client/Packages/com.beamable.server/Editor/Usam/Editor/CodeService.cs

```diff
@@ -405,74 +405,60 @@ public async Promise UpdateServicesVersions()
 		public async Promise RefreshServices()
 		{
 			ServiceDefinitions.Clear();
-			Promise finishedLocal = new Promise();
-			Promise finishedRemote = new Promise();
-
-			var promises = new List<Promise<Unit>>();
-			promises.Add(finishedLocal);
-			promises.Add(finishedRemote);
+			Promise finishedPopulatingServices = new Promise();
 
 			try
 			{
-				UsamLogger.Log("refresh remote services from CLI start");
+				UsamLogger.Log("refresh services from CLI start");
 				//Get remote only information from the CLI
-				var psRemote = _cli.ServicesPs(new ServicesPsArgs() { json = false, remote = true });
+				var psRemote = _cli.ServicesPs(new ServicesPsArgs() { json = false});
 				psRemote.OnStreamServiceListResult(cb =>
 				{
 					IsDockerRunning = cb.data.IsDockerRunning;
-					UsamLogger.Log($"Found {cb.data.BeamoIds.Count} remote services");
+					UsamLogger.Log($"Found {cb.data.BeamoIds.Count} services");
 					PopulateDataWithRemote(cb.data);
-					finishedLocal.CompleteSuccess();
+					finishedPopulatingServices.CompleteSuccess();
 				});
 				await psRemote.Run();
-				UsamLogger.Log("refresh remote services from CLI end");
-
-				UsamLogger.Log("refresh local services from CLI start");
-				//Get local only information from the CLI
-				var psLocal = _cli.ServicesPs(new ServicesPsArgs() { json = false, remote = false });
-				psLocal.OnStreamServiceListResult(cb =>
-				{
-					IsDockerRunning = cb.data.IsDockerRunning;
-					UsamLogger.Log($"Found {cb.data.BeamoIds.Count} local services");
-					PopulateDataWithRemote(cb.data);
-					finishedRemote.CompleteSuccess();
-				});
-				await psLocal.Run();
-				UsamLogger.Log("refresh local services from CLI end");
+				UsamLogger.Log("refresh services from CLI end");
 			}
 			catch
 			{
 				IsDockerRunning = false;
-				UsamLogger.Log("ERROR: Could not list remote services, skip");
+				UsamLogger.Log("ERROR: Could not list services, skip", true);
 				return;
 			}
 
-			var sequence = Promise.Sequence(promises);
-			await sequence;
+			await finishedPopulatingServices;
 		}
 
 		private void PopulateDataWithRemote(BeamServiceListResult objData)
 		{
 			for (int i = 0; i < objData.BeamoIds.Count; i++)
 			{
+				if (!objData.IsLocal[i] && !objData.IsRunningRemotely[i]) //In this case the service was disabled in the remote, which means that it's irrelevant.

```

Comment:
is this actually true? If something is not local, but it does exist remotely, even if its not running remotely, is that not helpful info to have?

Reply:
We don't have a `isArchived` flag, so the information we have is if it's local, running and enable on remote. I could check if it's enable or not, but I believe that is the same as if it's running or not? I believe that showing services that are not running and are possibly just archived is not very helpful information, seems more like something that I want to check in portal, instead of in the Unity Editor where I can't change it anyway.

If it exists locally, then I can actually change if it should be enable or not, and that is going to make it run remotely, but if it's not there, then why?

Reply:
fair enough!